### PR TITLE
Update cppitertools for conan v2

### PR DESCRIPTION
**Pull request by Ignition**
Specify library name and version:  **cppitertools/2.1**

I'm updating for it to be conan v2 compatabile

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by memsharded — DISMISSED** [review]
Thanks for updating this recipe!

File: recipes/cppitertools/all/conanfile.py
Comment (RubenRBS) [inline recipes/cppitertools/all/conanfile.py:57]: Does the recipe only support this version of boost or can it use a newer one?

**Review by RubenRBS — COMMENTED** [review]
Thanks a lot, only 1 small question before merging :)

File: recipes/cppitertools/all/conanfile.py
Comment (SpaceIm) [inline recipes/cppitertools/all/conanfile.py:31]: Could you add a layout please and use `get(self, **self.conan_data["sources"][self.version], strip_root=True` instead?

File: recipes/cppitertools/all/conanfile.py
Comment (SpaceIm) [inline recipes/cppitertools/all/conanfile.py:64]:
```suggestion
        self.info.clear()
```

File: recipes/cppitertools/all/conanfile.py
Comment (SpaceIm) [inline recipes/cppitertools/all/conanfile.py:8]:
```suggestion
required_conan_version = ">=1.50.0"
```

File: recipes/cppitertools/all/conanfile.py
Comment (SpaceIm) [inline recipes/cppitertools/all/conanfile.py:62]:
```suggestion
    def package_info(self):
        self.cpp_info.set_property("cmake_file_name", "cppitertools")
        self.cpp_info.set_property("cmake_target_name", "cppitertools::cppitertools")
```

Keep this information please, even if it matches default behavior of CMakeDeps, it gives a hint to maintainers that upstream provides CMake config file and recipe properly mimics its name and targets.

File: recipes/cppitertools/all/test_package/conanfile.py
Comment (SpaceIm) [inline recipes/cppitertools/all/test_package/conanfile.py:37]: There is a trick based on ctest to avoid this issue, see boost recipe for inspiration

File: recipes/cppitertools/all/conanfile.py
Comment (RubenRBS) [reply recipes/cppitertools/all/conanfile.py:38]: Good catch!

**Review by SSE4 — APPROVED** [review]

**Review by memsharded — APPROVED** [review]

**Review by RubenRBS — APPROVED** [review]

**Review by RubenRBS — APPROVED** [review]
Extra changes can go in a follow-up review :)

**Pull request merged**
